File: ocaml/rna-transcription/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let rna_complement dna =
    match dna with
    | `G -> `C
    | `C -> `G
    | `T -> `A
    | `A -> `U

let to_rna dna_list =
    List.map rna_complement dna_list

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn get-complement [nuc]
  (cond
    (= nuc \G) \C
    (= nuc \C) \G
    (= nuc \T) \A
    (= nuc \A) \U
    :else (throw (AssertionError. "Not a valid nucleotide"))))

(defn to-rna [dna]
  (apply str (map get-complement (char-array dna))))

